calicoctl/startup.py

```diff
@@ -255,15 +255,16 @@ def main():
     # Verify that the chosen IP exists on the current host
     warn_if_unknown_ip(ip, ip6)
 
-    # Set up etcd
-    ipv4_pools = client.get_ip_pools(4)
-    ipv6_pools = client.get_ip_pools(6)
-
-    # Create default pools if required
-    if not ipv4_pools:
-        client.add_ip_pool(4, DEFAULT_IPV4_POOL)
-    if not ipv6_pools and ipv6_enabled():
-        client.add_ip_pool(6, DEFAULT_IPV6_POOL)
+    if not os.getenv("NO_DEFAULT_POOLS"):
```

Comment:
I'd probably (over?)engineer this to require particular values to be set.  Sometimes it's easier for a wrapper to always set an environment variable but to set it to, say, "true" or "false" dynamically.